### PR TITLE
ci: add test runs to PRs

### PR DESCRIPTION
**Pull request by VerteDinde**
This PR adds a GH Actions test run for PRs.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macOS-latest, ubuntu-latest, windows-latest ]
+        arch: [ x64 ]
+        include:
+          - os: macOS-latest
+            arch: arm64
+    steps:
+      - name: Fix git checkout line endings
```
Comment (erickzhao) [inline .github/workflows/ci.yml:23]: Do we want to run lint only once on Ubuntu rather than hacking the line endings on Windows?

Comment (VerteDinde) [reply .github/workflows/ci.yml:23]: Yeah, that's probably a better approach, I'll amend this 🙇‍♀️

**Review by erickzhao — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            arch: arm64
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v3.4.1
+        with:
+          node-version: 16.x
+          cache: yarn
+      - name: Install
+        run: yarn install
```
Comment (aryanshridhar) [inline .github/workflows/ci.yml:30]: Not sure how crucial is this, but the docs [recommend](https://yarnpkg.com/cli/install#options-immutable) (The para below it explains better about --immutable) using [`yarn install --frozen-lockfile`](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-frozen-lockfile)/`yarn install --immutable` when working with CI?

Comment (erickzhao) [reply .github/workflows/ci.yml:30]: @aryanshridhar this is true, we can add that in a separate PR!

**Pull request merged**